app/script/calling/entities/FlowEntity.js

```diff
@@ -856,24 +855,47 @@ z.calling.entities.FlowEntity = class FlowEntity {
 
         const isModeDefault = this.negotiationMode() === z.calling.enum.SDP_NEGOTIATION_MODE.DEFAULT;
         if (isModeDefault && sendingOnTimeout) {
-          if (!this._containsRelayCandidate(iceCandidates)) {
-            const logMessage = `No relay ICE candidates in local SDP. Timeout reset\n${iceCandidates}`;
-            this.callLogger.warn(logMessage, iceCandidates);
-            return this._setSendSdpTimeout(false);
+          const MIN_NUMBER_OF_RELAY_SERVERS = 2;
+          if (this._getNumberOfRelayCandidates(iceCandidates) < MIN_NUMBER_OF_RELAY_SERVERS) {
+            const logMessage = `Less than ${MIN_NUMBER_OF_RELAY_SERVERS} relay ICE candidates in local SDP. Timeout reset\n${iceCandidates}`;
+            this.callLogger.warn(logMessage);
+            return this._setSendSdpTimeout();
           }
         }
 
+        const iceCandidateTypes = iceCandidates.reduce((types, candidateStr) => {
+          const typeMatches = candidateStr.match(/typ (\w+)/);
+          if (!typeMatches) {
+            return types;
+          }
+          const candidateType = typeMatches[1];
+          types[candidateType] = types[candidateType] ? types[candidateType] + 1 : 1;
+          return types;
+        }, {});
+
+        const iceCandidateTypesLog = Object.keys(iceCandidateTypes)
+          .map(candidateType => {
+            return `${iceCandidateTypes[candidateType]} ${candidateType}`;
+          })
+          .join(', ');
+
         const logMessage = {
           data: {
-            default: [localSdp.type, iceCandidates.length, this.remoteUser.name(), this.localSdp().sdp],
+            default: [
+              localSdp.type,
+              iceCandidates.length,
+              iceCandidateTypesLog,
+              this.remoteUser.name(),
+              this.localSdp().sdp,
+            ],
             obfuscated: [
               localSdp.type,
               iceCandidates.length,
               this.callLogger.obfuscate(this.remoteUser.id),
               this.callLogger.obfuscateSdp(this.localSdp().sdp),
             ],
           },
-          message: `Sending local '{0}' SDP containing '{1}' ICE candidates for flow with '{2}'\n{3}`,
+          message: `Sending local '{0}' SDP containing '{1}' ICE candidates ('{2}') for flow with '{3}'\n{4}`,
```

Comment:
Let's get rid of the `'` from within `('{2}')` as it would be a double indication of the expected variable content.